components/shared/form/BasicNumberInput.vue

```diff
@@ -21,30 +21,35 @@
   </NeoField>
 </template>
 
-<script lang="ts">
-import { Component, Prop, VModel, Vue } from 'nuxt-property-decorator'
+<script lang="ts" setup>
 import { NeoField, NeoInput } from '@kodadot1/brick'
 
-@Component({
-  components: {
-    NeoField,
-    NeoInput,
-  },
+const props = withDefaults(
+  defineProps<{
+    value: number
+    label: string
+    placeholder?: string
+    expanded?: boolean
+    message?: string
+    max?: number
+    min?: number
+    step?: number
+    minStep?: number
+  }>(),
+  {
+    expanded: false,
+    min: 1,
+    step: 1,
+  }
+)
+const emit = defineEmits(['input'])
+
+const vValue = computed({
+  get: () => props.value,
+  set: (value: number) => emit('input', value),
```

Comment:
[useVmodel](https://vueuse.org/core/useVModel/)

Reply:
Hello @roiLeo ,

I think we can use useVModel here because based on the documentation ![image](https://github.com/kodadot/nft-gallery/assets/3810177/8e45860b-c9f4-4a2c-9f45-310a6f34792f)
It will emit `update:modelValue` which in our case is `update:value` but we want to emit `input` instead.

Please let me know how i can use this here as i might have overlooked other ways to use useVModel

Reply:
> but we want to emit input instead.

@stephenjason89 I found the solution: `useVModel(props, 'value', emit, { eventName:  'input' })` can trigger correct event.